packages/skills/weather_client/strategy.py

```diff
@@ -0,0 +1,111 @@
+
+    def __init__(self, **kwargs) -> None:
+        """
+        Initialize the strategy of the agent.
+
+        :return: None
+        """
+        self._country = kwargs.pop('country') if 'country' in kwargs.keys() else DEFAULT_COUNTRY
+        self._search_interval = cast(float, kwargs.pop('search_interval')) if 'search_interval' in kwargs.keys() else DEFAULT_SEARCH_INTERVAL
+        self._max_row_price = kwargs.pop('max_row_price') if 'max_row_price' in kwargs.keys() else DEFAULT_MAX_ROW_PRICE
+        self.max_buyer_tx_fee = kwargs.pop('max_tx_fee') if 'max_tx_fee' in kwargs.keys() else DEFAULT_MAX_TX_FEE
+        self._currency_pbk = kwargs.pop('currency_pbk') if 'currency_pbk' in kwargs.keys() else DEFAULT_CURRENCY_PBK
+        self._ledger_id = kwargs.pop('ledger_id') if 'ledger_id' in kwargs.keys() else DEFAULT_LEDGER_ID
+        super().__init__(**kwargs)
+        self._search_id = 0
+        self.is_searching = True
+        self._last_search_time = datetime.datetime.now()
+
+    def get_next_search_id(self) -> int:
+        """
+        Get the next search id and set the search time.
+
+        :return: the next search id
+        """
+        self._search_id += 1
+        self._last_search_time = datetime.datetime.now()
+        return self._search_id
+
+    def get_service_query(self) -> Query:
+        """
+        Get the service query of the agent.
+
+        :return: the query
+        """
+        query = Query([Constraint(SEARCH_TERM, ConstraintType("==", self._country))], model=None)
+        return query
+
+    def is_time_to_search(self) -> bool:
+        """
+        Check whether it is time to search.
+
+        :return: whether it is time to search
+        """
+        if not self.is_searching:
+            return False
+        now = datetime.datetime.now()
+        diff = now - self._last_search_time
+        result = diff.total_seconds() > self._search_interval
+        return result
+
+    def is_acceptable_proposal(self, proposal: Description) -> bool:
+        """
+        Check whether it is an acceptable proposal.
+
+        :return: whether it is acceptable
+        """
+        result = (proposal.values['price'] - proposal.values['seller_tx_fee'] > 0) and \
+            (proposal.values['price'] <= self._max_row_price * proposal.values['rows']) and \
+            (proposal.values['currency_pbk'] == self._currency_pbk) and \
+            (proposal.values['ledger_id'] == self._ledger_id)
+        return result
+
+    def is_affordable_proposal(self, proposal: Description) -> bool:
+        """
+        Check whether it is an affordable proposal.
+
+        :return: whether it is affordable
+        """
+        payable = proposal.values['price'] + self.max_buyer_tx_fee
+        ledger_id = proposal.values['ledger_id']
+        address = cast(str, self.context.agent_addresses.get(ledger_id))
+        balance = self.context.ledger_apis.token_balance(ledger_id, address)
```

Comment:
balance should be set arbitrarily high, no ledger use!

Reply:
Removed the function since I am not checking if it affordable.